lib/temporal.rb

```diff
@@ -81,6 +83,59 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
+    # Long polls for a workflow to be completed and returns whatever the execute function
+    # returned.
+    # run_id of nil: await the entire workflow completion.  This can span multiple runs
+    # in the case where the workflow uses continue as new.
+    def await_workflow_result(workflow, workflow_id:, run_id: nil, **args)
```

Comment:
As far as refining the interface, it seems important that we use the word `await` (ish) so that people know this is a blocking call.
* I looked futures for activities, but it's all built into WorkflowContext and can't be used here.
* The assumption inside workflows is that things are blocking.  The assumption outside is that they are not.  So I don't think we should share the naming or approach we use when awaiting activities.

Reply:
We can easily add a non-blocking version of this that returns a future later on.